test/ectest.c

```diff
@@ -1159,6 +1159,40 @@ static int internal_curve_test_method(int n)
     return r;
 }
 
+static int group_field_test(void)
+{
+    int r = 1;
+    BIGNUM *sect521r1_field = NULL;
+    BIGNUM *sect163r2_field = NULL;
+    EC_GROUP *sect521r1_group = NULL;
+    EC_GROUP *sect163r2_group = NULL;
+
+    BN_hex2bn(&sect521r1_field,
+                "01FFFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
+                "FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
+                "FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
+                "FFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF"
+                "FFFF");
+
+
+    BN_hex2bn(&sect163r2_field,
+                "08000000000000000000000000000000"
+                "00000000C9");
+
+    sect521r1_group = EC_GROUP_new_by_curve_name(NID_secp521r1);
+    sect163r2_group = EC_GROUP_new_by_curve_name(NID_sect163r2);
```

Comment:
This will fail if no-ec2m is defined (this curve is a binary curve). It needs an `OPENSSL_NO_EC2M` guard :

````
$ ./config -d --strict-warnings no-ec2m
$ make
$ make TESTS=test_ec test
...
Test Summary Report
-------------------
../test/recipes/15-test_ec.t (Wstat: 256 Tests: 5 Failed: 1)
  Failed test:  2
  Non-zero exit status: 1
Files=1, Tests=5,  1 wallclock secs ( 0.02 usr  0.00 sys +  1.06 cusr  0.06 csys =  1.14 CPU)
Result: FAIL
Makefile:197: recipe for target '_tests' failed
make[1]: *** [_tests] Error 1
make[1]: Leaving directory '/home/matt/dev/openssl-write'
Makefile:195: recipe for target 'tests' failed
make: *** [tests] Error 2
````

Reply:
I added the guard for the relevant part of the test.